### PR TITLE
refactor corecheck boilerplate in CheckBase class

### PR DESCRIPTION
**Pull request by xvello**
### What does this PR do?

Create a `corecheck.CheckBase` class that corechecks can embed to get sane defaults for most of the `check.Check` interface. This removes lots of non-check-logic from the corecheck files for better readability, and will make changes easier in the future.

Also, this class provides a [different implementation of GetWarnings](https://github.com/DataDog/datadog-agent/blob/500cb7e7064462d09c4872e8eb07bc78373d1979/pkg/collector/corechecks/checkbase.go#L90) that avoids churning the `latestWarnings` array when it's empty.

As a side effect, the `warn`/`warnf` methods become public `Warn`/`Warnf` methods to be used by the embedding class. This made me realise only the `kubernetes_apiserver` uses that mechanism, I'll need to update the `docker` corecheck.

As they might change in the near future, I didn't update the `embed` corechecks, please let me know if that's desirable.

### CODE REVIEW

File: pkg/collector/check/id.go [hunk 1]
```diff
@@ -13,13 +13,18 @@ import (
 // ID is the representation of the unique ID of a Check instance
 type ID string
 
-// Identify returns the ID of the check
+// Identify returns a unique ID for a check and its configuration
```
Comment (JulienBalestra) [inline pkg/collector/check/id.go]: s/a/an

File: pkg/collector/check/id.go [hunk 2]
```diff
@@ -13,13 +13,18 @@ import (
...
+
+// BuildID returns a unique ID for a check name and its configuration
+func BuildID(checkName string, instance, initConfig ConfigData) ID {
 	h := fnv.New64()
 	h.Write([]byte(instance))
 	h.Write([]byte(initConfig))
 
-	id := check.String() + ":"
+	id := checkName + ":"
 	id += strconv.FormatUint(h.Sum64(), 16)
```
Comment (JulienBalestra) [inline pkg/collector/check/id.go]: Why not using a text formatting here like:
```golang
fmt.Sprintf("%s:%s", checkName, strconv.FormatUint(h.Sum64(), 16))
```

File: pkg/collector/corechecks/network/ntp.go [hunk 3]
```diff
@@ -53,7 +53,7 @@ func (c *NTPCheck) String() string {
 	return "ntp"
 }
 
-func (c *ntpConfig) Parse(data []byte, initData []byte) error {
+func (c *ntpConfig) parse(data []byte, initData []byte) error {
```
Comment (JulienBalestra) [inline pkg/collector/corechecks/network/ntp.go]: :+1:

**Review by JulienBalestra — CHANGES_REQUESTED** [review]
Pretty cool DRY idea !
Just few nits

**Review by masci — APPROVED** [review]
💯

**Review by JulienBalestra — APPROVED** [review]
:+1:

**Pull request merged**